subprojects/core/src/main/java/org/gradle/internal/progress/ProgressFormatter.java

```diff
@@ -15,7 +15,10 @@
  */
 package org.gradle.internal.progress;
 
+import org.gradle.api.Nullable;
+
 interface ProgressFormatter {
     String incrementAndGetProgress();
     String getProgress();
+    @Nullable String getProgressIndicator();
```

Comment:
I have change that such that `BuildProgressLogger` is no longer using any `ProgressFormatter`. Instead the `ProgressEvent` now contains the information about the progress (current, total, failing).
`ProgressBar` is now managed by `BuildStatusRenderer `.

We could probably split up `ProgressEvent`:
(1) events that render on the daemon side and only have a status (as it was before)
(2) events that are rendered on the client side and contain the actual progress information (as we do for the build progress)

Not sure if this has more implications and if we need to do that in this PR. It might not be worth the effort if we want to do more improvements of how build operation progress is rendered in general, as done in your spike, anyway.